doc/source/dev/contributor/quickstart_docker.rst

```diff
@@ -0,0 +1,177 @@
+:orphan:
+
+.. _quickstart-docker:
+
+=================================================
+Development Environment Quickstart Guide (Docker)
+=================================================
+
+This document shows how to start developing SciPy in a Docker container.
+These instructions should be considered a work in progress.
+
+Docker
+------
+
+Docker is a program for running Linux virtual machines within a host
+operating system. According to the `Docker website`_:
+
+ A Docker container image is a lightweight, standalone, executable package of
+ software that includes everything needed to run an application: code, runtime,
+ system tools, system libraries and settings.
+ Container images become containers at runtime and in the case of Docker
+ containers - images become containers when they run on Docker Engine.
+ Available for both Linux and Windows-based applications, containerized
+ software will always run the same, regardless of the infrastructure.
+
+Docker makes setting up a development environment easy and reliable: we
+provide a Docker image with suitable compilers already installed; you
+use the Docker engine to execute the image as a container, add the latest
+development version of SciPy and its build-time dependencies, and build
+SciPy.
+There are Docker hosts for several OS’s including
+macOS, Linux, and Windows; please follow the appropriate
+installation instructions for your operating system at `docs.docker.com`_.
+
+.. note::
+
+   If you have a version of an operating system that doesn't meet the
+   requirements of Docker Desktop, such as Windows 10 Home,
+   try `Docker Toolbox`_ .
+
+Cloning SciPy
+-------------
+
+Before starting SciPy's docker container, you should create a copy of the
+SciPy source code on your computer. That way, you'll be able to access the
+same files both from your native operating system and within the container.
+
+*Note: below we will use* terminal window *as a
+collective term that includes the Windows Command Prompt.*
+
+#. Browse to the `SciPy repository on GitHub`_ and `create your own fork`_.
+   You'll need to create a GitHub account if you don’t
+   already have one.
+#. Browse to your fork. Your fork will have a URL like
+   https://github.com/andyfaff/scipy, except with your GitHub username
+   in place of "andyfaff".
+#. Click the big, green "Clone or download" button, and copy the ".git"
+   URL to the clipboard. The URL will be the same as your fork’s url,
+   except it will end in ".git".
+#. Create a folder for the SciPy source code in a convenient place on
+   your computer. `Navigate`_ to it in the terminal window.
+#. Enter the command ``git clone`` followed by your fork’s .git URL.
+   Note that this creates in the terminal’s working directory a
+   ``scipy`` folder containing the SciPy source code. This assumes that
+   you have a ``git`` command line client that is available on your
+   PATH; if not, you can follow these `instructions to install a git client`_.
```

Comment:
There is a git client in the docker image, so an alternative could be `git clone` in the container. That might make subsequent steps more complicated.

Reply:
Yeah. I think installing git in host OS is OK. If they didn't, they'd still have to create a folder that can be shared between the host OS and docker (if they want access to the SciPy files from the host OS), right?

Reply:
>  If they didn't, they'd still have to create or choose a folder that can be shared between the host OS and docker (if they want access to the SciPy files from the host OS), right?

Yes.

If they don't install git on the host then they would have to fire up the container everytime they wanted to use git to push/pull from github..